### DISCUSSION TITLE
Is there a way to configure spotbugs for Spock tests ?

### DISCUSSION THREAD

**Discussion by walshe** [spockframework/spock#2009] [Q&A]
I have a gradle multiproject written in Java and am introducing Spock for unit testing. 

Spotbugs however goes wild on my Spock test files, so I've had to exclude them for now - see below

(  excludeFilter.set(rootProject.file("config/spotbugs/ignore.xml")) //exclude groovy code for now  )

````
HelloSpockSpec.setMetaClass(MetaClass) may expose internal representation by storing an externally mutable object into HelloSpockSpec.metaClass
This code stores a reference to an externally mutable object into the internal representation of the object.  If instances are accessed by untrusted code, and unchecked changes to the mutable object would compromise security or other important properties, you will need to do something different. Storing a copy of the object is better approach in many situations.
````

````
import spock.lang.Narrative
import spock.lang.Specification

@Narrative("Using a Spock table for parameterized test")
class HelloSpockSpec extends Specification {
	def "length of Spock's and his friends' names"() {

		expect:
		name.size() == length

		where:
		name     | length
		"Spock"  | 5
		"Kirk"   | 4
		"Scotty" | 6
	}
}

````

Does anyone have some sort of spotbugs config I could use ?

**Comment by Vampire** [comment]
That's not really Spock-specific.
The `setMetaClass` method is not something Spock adds.
This is something that Groovy generates during compilation.
So this should be the same with any Groovy class.
Go to https://groovyconsole.dev/?g=groovy_4_0&codez=eNpLzkksLlbwSM3JyQ8uyE_ODi5ITVao5qoFAHF0CNg and click "Inspect AST".

**Comment by efenderbosch-atg** [comment]
We use CodeNarc, w/ most of the Grails related rules turned off.

https://codenarc.org/
